src/geometry/euclidean.lean

```diff
@@ -920,4 +920,305 @@ calc dist (r1 • v +ᵥ p1) (r2 • v +ᵥ p2) * dist (r1 • v +ᵥ p1) (r2 
+        dist_comm, ←dist_eq_norm_vsub V p,
+        real.mul_self_sqrt (add_nonneg (mul_self_nonneg _) (mul_self_nonneg _))] at hcr₃,
+    change x * x + _ * (y * y) = _ at hcr₃,
+    rw [(show x * x + (1 - t₃) * (1 - t₃) * (y * y) =
+           x * x + y * y - 2 * y * (t₃ * y) + t₃ * y * (t₃ * y), by ring), add_left_inj] at hcr₃,
+    have ht₃ : t₃ = ycc₂ / y,
+    { field_simp [←hcr₃, hy0],
+      ring },
+    subst ht₃,
+    change cc₃ = cc₂ at hcc₃,
+    congr',
+    rw hcr₃val,
+    congr' 2,
+    field_simp [hy0],
+    ring }
+end
+
+/-- Given a finite nonempty affinely independent family of points,
+there is a unique (circumcenter, circumradius) pair for those points
+in the affine subspace they span. -/
+lemma exists_unique_dist_eq_of_affine_independent {ι : Type*} [hne : nonempty ι] [fintype ι]
+    {p : ι → P} (ha : affine_independent ℝ p) :
+  ∃! cccr : (P × ℝ), cccr.fst ∈ affine_span ℝ (set.range p) ∧
+    ∀ i, dist (p i) cccr.fst = cccr.snd :=
+begin
+  generalize' hn : fintype.card ι = n,
+  unfreezingI { induction n with m hm generalizing ι },
+  { exfalso,
+    have h := fintype.card_pos_iff.2 hne,
+    rw hn at h,
+    exact lt_irrefl 0 h },
+  { cases m,
+    { rw fintype.card_eq_one_iff at hn,
+      cases hn with i hi,
+      haveI : unique ι := ⟨⟨i⟩, hi⟩,
+      use (p i, 0),
+      simp only [prod.fst, prod.snd, set.range_unique, affine_subspace.mem_affine_span_singleton],
+      split,
+      { simp_rw [hi (default ι)],
+        use rfl,
+        intro i1,
+        rw hi i1,
+        exact dist_self _ },
+      { rintros ⟨cc, cr⟩,
+        simp only [prod.fst, prod.snd],
+        rintros ⟨rfl, hdist⟩,
+        rw hi (default ι),
+        congr',
+        rw ←hdist (default ι),
+        exact dist_self _ } },
+    { have i := hne.some,
+      let ι2 := {x // x ≠ i},
+      have hc : fintype.card ι2 = m + 1,
+      { rw fintype.card_of_subtype (finset.univ.filter (λ x, x ≠ i)),
+        { rw finset.filter_not,
+          simp_rw eq_comm,
+          rw [finset.filter_eq, if_pos (finset.mem_univ _),
+              finset.card_sdiff (finset.subset_univ _), finset.card_singleton, finset.card_univ,
+              hn],
+          simp },
+        { simp } },
+      haveI : nonempty ι2 := fintype.card_pos_iff.1 (hc.symm ▸ nat.zero_lt_succ _),
+      have ha2 : affine_independent ℝ (λ i2 : ι2, p i2) :=
+        affine_independent_subtype_of_affine_independent ha _,
+      replace hm := hm ha2 hc,
+      have hr : set.range p = insert (p i) (set.range (λ i2 : ι2, p i2)),
+      { change _ = insert _ (set.range (λ i2 : {x | x ≠ i}, p i2)),
+        rw [←set.image_eq_range, ←set.image_univ, ←set.image_insert_eq],
+        congr,
+        ext,
+        simp [classical.em] },
+      change ∃! (cccr : P × ℝ), (_ ∧ ∀ i2, (λ q, dist q cccr.fst = cccr.snd) (p i2)),
+      conv {
+        congr,
+        funext,
+        conv { congr, skip, rw ←set.forall_range_iff }
+      },
```

Comment:
Hmm... seems like there are some formatting issues...
Please take a look at those

bors r-

Reply:
I've made some changes to `conv` formatting there, is that what's intended?